bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,323 @@
+                }
+                break;
+            case CHANNEL_INPUT:
+                if (command == RefreshType.REFRESH) {
+                    StringType input = new StringType(device.getInputStatus().getResult().getValue());
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_INPUT, input);
+                } else {
+                    logger.trace("Received input command" + command);
+                    Input input = new Input(((StringType) command).toString());
+                    device.setInput(input);
+                }
+                break;
+            case CHANNEL_AUDIO_MUTE:
+            case CHANNEL_VIDEO_MUTE:
+                boolean isAudioMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE);
+                boolean isVideoMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE);
+                if (isVideoMute || isAudioMute) {
+                    if (command == RefreshType.REFRESH) {
+                        // refresh both video and audio mute, as it's one request
+                        MuteQueryResponseValue muteStatus = device.getMuteStatus();
+                        updateState(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE,
+                                muteStatus.isAudioMuted() ? OnOffType.ON : OnOffType.OFF);
+                        updateState(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE,
+                                muteStatus.isVideoMuted() ? OnOffType.ON : OnOffType.OFF);
+                    } else {
+                        if (isAudioMute) {
+                            logger.trace("Received audio mute command" + command);
+                            boolean muteOn = command == OnOffType.ON;
+                            device.setMute(MuteInstructionChannel.AUDIO, muteOn);
+                        }
+                        if (isVideoMute) {
+                            logger.trace("Received video mute command" + command);
+                            boolean muteOn = command == OnOffType.ON;
+                            device.setMute(MuteInstructionChannel.VIDEO, muteOn);
+                        }
+                    }
+                }
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+            logger.trace("Successfully handled command {} on channel {}", command, channelUID.getId());
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (ResponseException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (ConfigurationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        } catch (AuthenticationException e) {
+            this.handleAuthenticationException(e);
+        }
+    }
+
+
+    @Override
+    public void initialize() {
+        try {
```

Comment:
Here is how I will do it to avoid a thing remaining in INITIALIZING state in case the setup takes too much time (probably only if the projector is not responding):
```
try {
    this.getDevice();
} catch (ConfigurationException e) {
    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
    return;
}
setupJob = scheduler.schedule() -> {
    setupDevice();
    setupRefreshInterval();  
}, 0, TimeUnit.SECONDS);
```
And of course you cancel setupJob in the dispose() method.

Reply:
Note that with my code, setupDevice() will never throw ConfigurationException because getDevice() will be called just before.

Reply:
There are other exception types that can happen during setupDevice() and setupRefreshInterval(), I decided to pull them up and handle all of them at the same level.